pkg/file/pipeline/hashtrie/hashtrie.go

```diff
@@ -245,5 +259,19 @@ func (h *hashTrieWriter) Sum() ([]byte, error) {
 
 	// return the hash in the highest level, that's all we need
 	data := h.buffer[0:h.cursors[maxLevel]]
-	return data[swarm.SpanSize:], nil
+	rootHash := data[swarm.SpanSize:]
+
+	// save disperse replicas of the root chunk
+	if h.rParams.Level() != redundancy.NONE {
+		rootData, err := h.rParams.GetRootData()
+		if err != nil {
+			return nil, err
+		}
+		putter := replicas.NewPutter(h.replicaPutter)
+		err = putter.Put(h.ctx, swarm.NewChunk(swarm.NewAddress(rootHash), rootData))
```

Comment:
here we should not use context from the `hashTrieWriter`. Either `Sum` should be extended with a context or in fact the whole integration is possibly at the wrong place. I think it should happen after and where Sum is called. There the putter/getter used underlyingly is known so its easy to then just wrap the putter in the replicas.Putter and use that to Put the chunk gotten from the getter.

Reply:
the redundancy cache is available in the hashtrie package from which the root chunk data is retrieved, so I would argue whether the integration is a great place.